internal/rbd/rbd_util.go

```diff
@@ -1071,6 +1106,29 @@ func (rv *rbdVolume) cloneRbdImageFromSnapshot(ctx context.Context, pSnapOpts *r
 		return fmt.Errorf("failed to create rbd clone: %w", err)
 	}
 
+	// delete the cloned image if a next step fails
+	deleteClone := true
+	defer func() {
+		if deleteClone {
+			err = librbd.RemoveImage(rv.ioctx, rv.RbdImageName)
```

Comment:
use rv.DeleteImage which uses manager tasks to delete the image?

Reply:
Not sure that is needed, it is only for an error case that the just created image needs to be removed again.

Because the `deleteImage()` helper uses CephMgr, it also needs access to the Credentials, which are not available here. I think it would clutter the code more than is useful.

Reply:
RemoveImage directly from the ceph is time-consuming that's why we brought a manager to handle the image deletion asynchronous. I think this is fine for now.